### ISSUE TITLE
[问题/Issue] 章节5：训练细节和tokenizer

### ISSUE THREAD

**Issue by luoclab** [datawhalechina/happy-llm#17]
### 1. 遇到问题的章节 / Affected Chapter

Chapter5

### 2. 具体问题描述 / Problem Description

1.作者最后的结果是训练一个epoch实现的吗？，是否可以提供训练好的权重提供给我进行验证
2.tokenizer.json文件的vocab里面是没有中文显示的，是否可以考虑可以清晰可视化中文的vocab，比如使用Whitespace / Char 分词器

### 3. 问题重现材料 / Reproduction Materials

第五章训练细节

### 确认事项 / Verification

- [x] 此问题未在过往Issue中被报告过 / This issue hasn't been reported before

**Comment by KMnO4-zx** [comment]
问题1: pretrain阶段经过一个epoch训练完成，sft阶段经过3个epoch完成。近期会准备一下之前的训练结果给大家做参考，也会公开训练权重，方便大家进行验证。

问题2:tokenizer使用预训练数据集进行训练，其中包含了大量的中文语料。

**Comment by luoclab** [comment]
问题1.可以解释一下targets为什么是“attention_mask”吗，为什么不是“label”?

![Image](https://github.com/user-attachments/assets/885750ab-50cf-4d05-aaa2-5013f2acaa01)
问题2.我的理解是应该在attention类过程中需要用到padding_mask,但是我没有看到padding_mask用在什么地方，
是在211行的scores = scores + self.mask[:, :, :seqlen, :seqlen]#屏蔽的位置填 -1e9吗
这个是三角掩码（遮住未来）还是padding_mask（遮住padding）,我不是很理解这部分代码，可以解答一下三角掩码（遮住未来）还是padding_mask（遮住padding）分别实现在那个地方吗，希望可以解答

**Comment by KMnO4-zx** [comment]
> 我看到了代码实现是在loss的时候把padding位置的loss置为0，但是这样就不用在每个layer算注意力的时候都设置padding_mask吗，我还在学习阶段，希望可以理解这些原理

是这样的，可以参照 [第五章dataset](https://github.com/datawhalechina/happy-llm/blob/main/docs/chapter5/%E7%AC%AC%E4%BA%94%E7%AB%A0%20%E5%8A%A8%E6%89%8B%E6%90%AD%E5%BB%BA%E5%A4%A7%E6%A8%A1%E5%9E%8B.md#532-dataset) dataset 中的讲解看一下～

**Comment by luoclab** [comment]
可以给我解答一下为什么不用在每个layer中都使用padding_mask吗

**Comment by KMnO4-zx** [comment]
https://github.com/datawhalechina/happy-llm/blob/main/docs/chapter5/code/dataset.py

可以详细理解一下dataset的设计
